patroni/postgresql/validator.py

```diff
@@ -12,6 +14,53 @@
 
 logger = logging.getLogger(__name__)
 
+if sys.version_info < (3, 9):
+    PathLikeObj = Path
+
+    def _get_validator_files() -> List[PathLikeObj]:
+        conf_dir = os.path.join(
+            os.path.dirname(os.path.abspath(__file__)),
+            'available_parameters',
+        )
+        yaml_files: List[PathLikeObj] = []
+
+        for root, _, filenames in os.walk(conf_dir):
+            for file in sorted(filenames):
+                full_path = os.path.join(root, file)
+                if file.lower().endswith(('.yml', '.yaml')):
+                    yaml_files.append(Path(full_path))
+                else:
+                    logger.info(
+                        'Ignored a non-YAML file found under `available_parameters` directory: `%s`.',
+                        full_path)
```

Comment:
There is quite a lot of duplicated code.
I think we should do the following instead:
1. yield Path-like objects from from both _get_validator_files() functions
2. create another function that will do filtering and sorting by name
3. we should silently skip `*.py` and `*.pyc` files.

On top of that I think we can move the entire if ... else block to the `available_parameters/__init__py`.

Please also add some doctrings to all new functions! :)